### PR TITLE
[NOD-530] feat: Semantic versioning setup

### PR DESCRIPTION
**Pull request by GiampieroPorfiri**
### [NOD-530] feat: Semantic versioning setup

This PR is crucial for updating the repository to utilize the new semver setup.

#### List of Changes
GH Action Version Update: The version of the GH Action setup for semantic versioning has been updated: nodo5-semver-setup@[SHA].

#### Motivation and Context

This PR aims to keep the GitHub Action updated.

#### How Has This Been Tested?

The change has been tested by running the pipeline.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

[NOD-530]: https://pagopa.atlassian.net/browse/NOD-530?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by GiampieroPorfiri** [comment]
update_code

**Comment by GiampieroPorfiri** [comment]
_update_code_

**Comment by GiampieroPorfiri** [comment]
*update_code*

File: .github/workflows/04_release_deploy.yml [hunk 1]
```diff
@@ -64,6 +64,12 @@ jobs:
...
       environment: ${{ steps.output.outputs.environment }}
     steps:
+      - name: Semver setup
+        id: semver_setup
+        uses: pagopa/github-actions-template/nodo5-semver-setup@ce252c8501c9242bd6045f7cdd650736b2f38777
+        with:
+          semver: ${{ inputs.semver }}
+
       # Check branch or tag
       - name: Checkout code
```
Comment (andrea-deri) [inline .github/workflows/04_release_deploy.yml:74]: in order to add `Semver setup` step, you must remove the other following steps, as done [here](https://github.com/pagopa/pagopa-fdr-json-to-xml/blob/bca008638cbb3802d76573843fa18145f118edbf/.github/workflows/04_release_deploy.yml#L61)

**Comment by GiampieroPorfiri** [comment]
*update_code*

**Pull request merged**